src/Controller/WopiController.php

```diff
@@ -278,4 +313,42 @@ public function wopi(string $action, string $id, Request $request) {
     return $response;
   }
 
+  /**
+   * Decodes and verifies a JWT token.
+   *
+   * Verification include:
+   *  - matching $id with fid in the payload
+   *  - verifying the expiration.
+   *
+   * @param string $token
+   *   The token to verify.
+   * @param int|string $expected_media_id
+   *   Media id expected to be in the token payload.
+   *   This could be a stringified integer like '123'.
+   *
+   * @return array|null
+   *   Data decoded from the token, or NULL on failure or if the token has
+   *   expired.
+   */
+  protected function verifyTokenForMediaId(
```

Comment:
Im thinking since we are decoding and verifing in the token manager keep the verify methods in the same place.
For me seems strange to have the same operation in the service and controller.

Reply:
The idea here is that the token manager should not care whether we are encoding media ids or something else.
We could change this design..

In fact we have 3 layers of functionality:
- The raw encode/decode.
- The timeout handling.
- The specific values relevant to a media entity.

This could give us different services each with their own interface..

Reply:
Actually yes, makes sense to keep them separated, adding all those services is too much.